### PR TITLE
[REVIEWS] fix_review_state

### PR DESCRIPTION
**Pull request by aurorahey**
#### What problem is this solving?

the previous fix about loading star causing infinite request, this PR fix both the infinite request and the PDP loading star problem. Also improved _GetAverageRatingByProductId_ query.

#### How to test it?

1. Use Incognito mode to open link
https://esika.tiendabelcorp.cl/vibranza-perfume-de-mujer-45-ml/p?workspace=aurora
you should see all the stars showing

2. open any review page to see the requests

### CODE REVIEW

File: react/Reviews.tsx
Comment (polishq) [inline react/Reviews.tsx:607]: Why not combine this with the `useEffect` above? They both have the same dependencies, so you could handle everything at the same time

**Review by polishq — CHANGES_REQUESTED** [review]

File: dotnet/Services/ProductReviewService.cs
Comment (btalma) [inline dotnet/Services/ProductReviewService.cs:140]: Before we were not including reviews with a zero rating in the total.  Do we know that we will will always have a rating of at least one?

**Review by polishq — APPROVED** [review]
React looks good
Wait for @btalma 's approval for the .NET

**Pull request closed without merging**
